### PR TITLE
Make `make_request` actually render the request

### PR DESCRIPTION
**Pull request by richvdh**
`make_request` previously stubbed out `request.process`; remove this stubbing so that `make_request` renders the request as well as building it.

(`request.requestReceived` calls `request.process`, which calls `site.getResourceFor` and `resource.render`).

There are a few separate commits here which should be reviewable independently. the `url_preview` changes are a preparatory commit to make sure that `site.getResourceFor` will return the `PreviewUrlResource`.

### CODE REVIEW

File: tests/rest/media/v1/test_media_storage.py [hunk 1]
```diff
@@ -359,7 +359,6 @@ def _test_thumbnail(self, method, expected_body, expected_found):
                 channel.json_body,
                 {
                     "errcode": "M_NOT_FOUND",
-                    "error": "Not found [b'example.com', b'12345?width=32&height=32&method=%s']"
```
Comment (clokep) [inline tests/rest/media/v1/test_media_storage.py:362]: What caused this to change? It doesn't seem like it should have...

**Review by clokep — APPROVED** [review]
LGTM.

File: tests/rest/media/v1/test_media_storage.py (hunk 1 above)
Comment (richvdh) [reply tests/rest/media/v1/test_media_storage.py:362]: this error is built from `request.postpath`. Previously, we were manually setting `postpath` in `make_request` [here](https://github.com/matrix-org/synapse/pull/8761/files#diff-b623bf737404eea7a58674e12e1f83601abfa4a97c3d0da62768ee263a9fbd06L232), based on the full path of the request we are simulating.

we're now leaving `request.process` to populate `postpath`, which it does [here](https://github.com/twisted/twisted/blob/268354f9319d772aba3c565e8b3c390e0440d3c6/src/twisted/web/server.py#L216) based on `request.path`, which is set [here](https://github.com/twisted/twisted/blob/268354f9319d772aba3c565e8b3c390e0440d3c6/src/twisted/web/http.py#L938) and excludes the query-string.

In short: it now better matches the real behaviour.

Comment (clokep) [reply tests/rest/media/v1/test_media_storage.py:362]: Makes sense! Thanks!
